roles/tas_single_node/tasks/podman/tsa.yml

```diff
@@ -0,0 +1,73 @@
+---
+- name: Confirmed required parameters provided
+  ansible.builtin.assert:
+    that:
+      - tas_single_node_base_hostname is defined
+      - tas_single_node_base_hostname | trim | length > 0
+    msg: "'tas_single_node_base_hostname' must be specified"
+
+- name: Slurp TSA Certificates chain
+  ansible.builtin.slurp:
+    src: "{{ item }}"
+  loop:
+    - "{{ tas_single_node_remote_tsa_private_key }}"
+    - "{{ tas_single_node_remote_tsa_certificate_chain }}"
+    - "{{ tas_single_node_remote_tsa_signer_private_key }}"
+  register: remote_tsa_certificates
+
+- name: Define TSA base secret data
+  ansible.builtin.set_fact:
+    secret_data:
+      private_key.pem: |
+        {{ (remote_tsa_certificates.results | selectattr('source', 'equalto', tas_single_node_remote_tsa_signer_private_key) | list | first).content }}
+      certificate-chain.pem: |
+        {{ (remote_tsa_certificates.results | selectattr('source', 'equalto', tas_single_node_remote_tsa_certificate_chain) | list | first).content }}
+      password: "{{ tas_single_node_tsa_signer_passphrase | b64encode }}"
+
+- name: Append Tink secret data
+  ansible.builtin.set_fact:
+    secret_data: "{{ secret_data | combine({'encrypted_key_set': tas_single_node_tsa_tink_keyset | b64encode}) }}"
+  when: tas_single_node_signer_type == 'tink'
+
+- name: Create TSA Secret
+  ansible.builtin.copy:
+    content: "{{ secret_content | to_nice_yaml(indent=2) }}"
+    dest: "{{ tas_single_node_tsa_secret }}"
+    mode: "0600"
+  vars:
+    secret_content:
+      kind: Secret
+      apiVersion: v1
+      metadata:
+        name: tsa-server-secret
+        namespace: tsa-system
+      data: "{{ secret_data }}"
+  register: tsa_secret_result
+
+- name: Ensure /manifests/tsa directory exists
+  ansible.builtin.file:
+    path: /manifests/tsa
+    state: directory
+    mode: "0700"
+
+- name: Render TSA Deployment YAML
+  ansible.builtin.template:
+    src: manifests/tsa/tsa-server.j2
+    dest: /manifests/tsa/tsa-server.yaml
+    mode: "0700"
+
+- name: Read the rendered YAML file
+  ansible.builtin.slurp:
+    src: /manifests/tsa/tsa-server.yaml
+  register: tsa_server_yaml
+
+- name: Deploy TSA Pod
+  ansible.builtin.include_tasks: podman/install_manifest.yml
+  vars:
+    podman_spec:
+      state: started
+      systemd_file: tsa
+      network: "{{ tas_single_node_podman_network }}"
+      kube_file_content: "{{ tsa_server_yaml.content | b64decode | from_yaml }}"
```

Comment:
For the other services, we currently use e.g. `"{{ lookup('ansible.builtin.template', 'manifests/nginx/nginx.yaml') | from_yaml }}"` without having to explicitly render the file on drive. Is there a reason why this wouldn't work here?

Reply:
In the other services, we don't use j2 templates as we are trying to do for TSA (we need conditionals) 
That's why here, we are rendering the j2 template into a yaml before using it.

Reply:
But isn't that what [template lookup](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_lookup.html) does? Sorry if I'm missing something, but I think we should be able to do here what other services do (I haven't tried this myself, so I could be wrong though)

Reply:
you're correct! Thanks!